Userland/Libraries/LibArchive/Zip.cpp

```diff
@@ -138,6 +140,63 @@ ErrorOr<void> ZipOutputStream::add_member(ZipMember const& member)
     return local_file_header.write(*m_stream);
 }
 
+ErrorOr<double> ZipOutputStream::add_member_from_stream(StringView path, Stream& stream, Optional<Core::DateTime> optional_date)
+{
+    auto buffer = TRY(stream.read_until_eof());
+
+    Archive::ZipMember member {};
+    member.name = TRY(String::from_utf8(path));
+
+    if (optional_date.has_value()) {
+        auto date = optional_date.value();
+        member.modification_date = to_packed_dos_date(date.year(), date.month(), date.day());
+        member.modification_time = to_packed_dos_time(date.hour(), date.minute(), date.second());
+    }
+
+    auto deflate_buffer = Compress::DeflateCompressor::compress_all(buffer);
+    auto compression_ratio = 0.0;
```

Comment:
Since you only update this value if `deflate_buffer.value().size() < buffer.size()`, shouldn't the default be `1.f`? (i.e. no compression)